subprojects/core/src/main/java/org/gradle/execution/TaskNameResolver.java

```diff
@@ -120,7 +121,7 @@ private static TaskInternal getExistingTask(ProjectInternal project, String task
     private void collectTaskNames(ProjectInternal project, Set<String> result) {
         discoverTasks(project);
         result.addAll(getTaskNames(project));
-        for (Project subProject : project.getChildProjects().values()) {
+        for (Project subProject : ProjectHierarchyUtils.getChildProjectsForInternalUse(project)) {
```

Comment:
💄 Static import `getChildProjectsForInternalUse` for more fluent code

Reply:
Good idea, thanks! Done.